main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/SourceCodeLookup.cs

```diff
@@ -120,49 +120,101 @@ public static FilePath TryDebugSourceFolders (FilePath originalFile, byte[] hash
 			return FilePath.Null;
 		}
 
-		public static bool CheckFileHash (FilePath file, byte[] hash)
+		static void ComputeHashes (Stream stream, HashAlgorithm hash, HashAlgorithm dos, HashAlgorithm unix)
 		{
-			if (hash == null)
-				return false;
-			if (File.Exists (file)) {
-				using (var fs = File.OpenRead (file)) {
-					// Roslyn SHA1 checksum always starts with 20
-					if (hash.Length > 0 && hash [0] == 20)
-						using (var sha1 = SHA1.Create ()) {
-							if (sha1.ComputeHash (fs).Take (15).SequenceEqual (hash.Skip (1))) {
-								return true;
-							}
-						}
-					if (hash.Length > 0 && hash [0] == 32)
-						using (var sha1 = SHA256.Create ()) {
-							if (sha1.ComputeHash (fs).Take (15).SequenceEqual (hash.Skip (1))) {
-								return true;
-							}
-						}
-					if (hash.Length == 20) {
-						using (var sha1 = SHA1.Create ()) {
-							fs.Position = 0;
-							if (sha1.ComputeHash (fs).SequenceEqual (hash)) {
-								return true;
-							}
-						}
-					}
-					if (hash.Length == 32) {
-						using (var sha256 = SHA256.Create ()) {
-							fs.Position = 0;
-							if (sha256.ComputeHash (fs).SequenceEqual (hash)) {
-								return true;
-							}
-						}
+			var unixBuffer = new byte[4096];
```

Comment:
```suggestion
			var pool = ArrayPool<byte>Shared;
			var unixBuffer = pool.Rent (4096);
```

And at the end of the method:

```csharp
			pool.Return (unixBuffer);
```

And same for the others.

Reply:
done